Odometry Docs (#406)
OkapiLib/OkapiLib: docs/tutorials/walkthrough/odometry.md

# Odometry

Odometry is the act of tracking a robot's absolute position. 
This information can be used by motion algorithms to drive to positions or turn to absolute angles.

An excellent overview of the odometry tracking algorithm can be found [here](https://www.vexforum.com/t/team-5225-introduction-to-position-tracking-document/49640).

> # Configuring Odometry

Reviewer: ```suggestion
## Configuring Odometry
```

> OkapiLib supports odometry for all chassis configurations, and is configured using 

Reviewer: Can you rephrase this? It sounds awkward.

Author: ```suggestion
OkapiLib supports odometry for all chassis configurations. A chassis controller that uses odometry is configured using the 
```
Better?

    // specify the middle encoder distance (1 in) and diameter (2.75 in)
    .withOdometry({{3_in, 7_in, 1_in, 2.75_in}, quadEncoderTPR})
    .buildOdometry()
```

Here is the same example using [ChassisControllerPID](@ref okapi::ChassisControllerPID):

```cpp
ChassisControllerBuilder()
    .withMotors(1, -2) // left motor is 1, right motor is 2 (reversed)
    .withGains(
        {0.001, 0, 0.0001}, // distance controller gains
        {0.001, 0, 0.0001}, // turn controller gains
        {0.001, 0, 0.0001}  // angle controller gains (helps drive straight)
    )
    .withSensors(
        {'A', 'B'}, // left encoder in ADI ports A & B
        {'C', 'D', true}  // right encoder in ADI ports C & D (reversed)
        {'E', 'F'}  // middle encoder in ADI ports E & F
    )
    // green gearset, 4 inch wheel diameter, 11 inch wheelbase
    // 1 inch middle encoder distance, and 2.75 inch middle wheel diameter
    .withDimensions(AbstractMotor::gearset::green, {{4_in, 11_in, 1_in, 2.75_in}, imev5GreenTPR})
    .withOdometry() // use the same scales as the chassis (above)
    .buildOdometry() // build an odometry chassis
```

> # Using Odometry

Reviewer: ```suggestion
## Using Odometry
```
